### ISSUE TITLE
[BUG] Possible issue with RTags not finding rc & rdm

### ISSUE THREAD

**Issue by ijacquez** [hlissner/doom-emacs#1864]
**Describe the issue**
Executables `rc` and `rdm` can't be found.

I'm suspecting it's related to the difference between `rtags-path` and `rtags-install-path`. From what I can see, `rtags-path` best suites our needs as `rtags` is set to be installed in `(concat doom-etc-dir "rtags/")`.

Also, shouldn't `(when (/= (length bins) 0)` be `(when (/= (length bins) 2)`?

<details>
In `modules/lang/cc/doctor.el`

```
(when (require 'rtags nil t)
  ;; rtags
  (let ((bins (cl-remove-if #'executable-find `(,rtags-rdm-binary-name ,rtags-rc-binary-name))))
    (when (/= (length bins) 0)
      (warn! "Couldn't find the rtag client and/or server programs %s. Disabling rtags support" bins))))
```
In `src/rtags.el`
```
(defun rtags-executable-find (exe)
  (cond ((and rtags-tramp-enabled (tramp-tramp-file-p default-directory)) exe)
        ;; for tramp let's rely on `tramp-remote-path`, so if You have some *debug*
        ;; directory to store RTags binaries, just put it to `tramp-remote-path`
        (rtags-path
         (or
          (let ((file (expand-file-name exe rtags-path)))
            (and (file-executable-p file) file))
          (let ((file (expand-file-name exe (concat rtags-path "/bin/"))))
            (and (file-executable-p file) file))))
        ((let* ((install-path (rtags-package-install-path))
                (file (and install-path (concat install-path "/rtags-" rtags-package-version "/bin/" exe))))
           (and file (file-executable-p file) file)))
        (t (executable-find exe))))
```
</details>

**Steps to reproduce**
Running `bin/doom doctor` should result in a warning regarding `rc` and `rdm` not being found.

**System information**
I'm not sure why `doom/info` is reporting `fatal: not a git repository (or any of the parent directories): .git`. It could be that I have a symbolic link from `doom-emacs` to `~/.emacs.d`.
<details>

```
emacs   version    26.1
        features   XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 THREADS LIBSYSTEMD LCMS2
        build      Apr 11, 2019
        buildopts  (--build x86_64-linux-gnu --prefix=/usr --sharedstatedir=/var/lib --libexecdir=/usr/lib --localstatedir=/var/lib --infodir=/usr/share/info --mandir=/usr/share/man --enable-libsystemd --with-pop=yes --enable-locallisppath=/etc/emacs:/usr/local/share/emacs/26.1/site-lisp:/usr/local/share/emacs/site-lisp:/usr/share/emacs/26.1/site-lisp:/usr/share/emacs/site-lisp --with-sound=alsa --without-gconf --with-mailutils --build x86_64-linux-gnu --prefix=/usr --sharedstatedir=/var/lib --libexecdir=/usr/lib --localstatedir=/var/lib --infodir=/usr/share/info --mandir=/usr/share/man --enable-libsystemd --with-pop=yes --enable-locallisppath=/etc/emacs:/usr/local/share/emacs/26.1/site-lisp:/usr/local/share/emacs/site-lisp:/usr/share/emacs/26.1/site-lisp:/usr/share/emacs/site-lisp --with-sound=alsa --without-gconf --with-mailutils --with-x=yes --with-x-toolkit=gtk3 --with-toolkit-scroll-bars 'CFLAGS=-g -O2 -fdebug-prefix-map=/build/emacs-cQxEw2/emacs-26.1+1=. -fstack-protector-strong -Wformat -Werror=format-security -Wall' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro')
        windowsys  x
        daemonp    server-running
doom    version    2.0.9
        build      fatal: not a git repository (or any of the parent directories): .git
system  type       gnu/linux
        config     x86_64-pc-linux-gnu
        shell      /usr/bin/fish
        uname      Linux 5.0.0-27-generic #28-Ubuntu SMP Tue Aug 20 19:53:07 UTC 2019 x86_64
        path       (~/.local/bin /snap/bin ~/x-tools/i686-w64-mingw32/bin ~/x-tools/i686-pc-linux-gnu/bin ~/x-tools/x86_64-w64-mingw32/bin ~/x-tools/sh2eb-elf/bin ~/.local/bin /snap/bin ~/x-tools/i686-w64-mingw32/bin ~/x-tools/i686-pc-linux-gnu/bin ~/x-tools/x86_64-w64-mingw32/bin ~/x-tools/sh2eb-elf/bin ~/.local/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games /snap/bin ~/.local/tool-chains/bin ~/.local/tool-chains/sh-elf/bin /usr/lib/emacs/26.1/x86_64-linux-gnu)
config  envfile    nil
        elc-files  0
        modules    (:completion (company +auto) ivy :ui workspaces doom doom-quit hl-todo modeline nav-flash ophints (popup +all +defaults) unicode vc-gutter vi-tilde-fringe window-select :editor snippets fold multiple-cursors rotate-text :emacs dired electric vc :tools (lookup +docsets) eval editorconfig flycheck flyspell magit rgb :lang assembly cc common-lisp csharp data emacs-lisp markdown (org +dragndrop +attach +babel +capture +export +present) python (sh +fish) :config (default +bindings +smartparens))
        packages   (n/a)
        elpa       (n/a)
```
</details>

P.S.: I'll be getting back to PR #1437. I got Irony up and running, so I see how RTags+Irony would be an ideal combo.

**Comment by hlissner** [comment]
Would this change be sufficient?

```diff
   ;; rtags
-  (let ((bins (cl-remove-if #'executable-find `(,rtags-rdm-binary-name ,rtags-rc-binary-name))))
+  (let ((bins (cl-remove-if #'rtags-executable-find `(,rtags-rdm-binary-name ,rtags-rc-binary-name))))
     (when (/= (length bins) 0)
```

> Also, shouldn't (when (/= (length bins) 0) be (when (/= (length bins) 2)?

No, because the `cl-remove-if` predicate is inverted. It will only remove binaries if they are found. This way, the following warning can complain about the list of executables that weren't found.

**Comment by hlissner** [comment]
This should be fixed as of 009cfbc. Let me know if that isn't the case and I'll reopen the issue. Thanks for bringing it to my attention!
